File: Chapter05/FS/EmpiricalFontSize/EmpiricalFontSize/EmpiricalFontSize/FontCalc.fs
namespace EmpiricalFontSize

open System
open Xamarin.Forms

type FontCalc(label : Label, fontSize : float, containerWidth : float) =

    // Recalculate the size of the Label with the specified font size
    do label.FontSize <- fontSize
    let sizeRequest = label.Measure(containerWidth, Double.PositiveInfinity)

    // Save the font size and the resultant Label height in public properties.
    member val FontSize = fontSize

    member val TextHeight = sizeRequest.Request.Height

File: Chapter05/EmpiricalFontSize/EmpiricalFontSize/EmpiricalFontSize/FontCalc.cs
using System;
using Xamarin.Forms;

namespace EmpiricalFontSize
{
    struct FontCalc
    {
        public FontCalc(Label label, double fontSize, double containerWidth)
            : this()
        {
            // Save the font size.
            FontSize = fontSize;

            // Recalculate the Label height.
            label.FontSize = fontSize;
            SizeRequest sizeRequest =
                label.Measure(containerWidth, Double.PositiveInfinity);

            // Save that height.
            TextHeight = sizeRequest.Request.Height;
        }

        public double FontSize { private set; get; }

        public double TextHeight { private set; get; }
    }
}
